From 44491edd34761fc052852399dfe214b5d8242d0f Mon Sep 17 00:00:00 2001
From: roshan-ku <roshan.kumar@intel.com>
Date: Wed, 8 Apr 2026 10:23:37 +0530
Subject: [PATCH 1/1] Patch mtl_st20p_tx for colour format additional support

---
 ecosystem/ffmpeg_plugin/mtl_st20p_tx.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/ecosystem/ffmpeg_plugin/mtl_st20p_tx.c b/ecosystem/ffmpeg_plugin/mtl_st20p_tx.c
index e182b38..d233a90 100644
--- a/ecosystem/ffmpeg_plugin/mtl_st20p_tx.c
+++ b/ecosystem/ffmpeg_plugin/mtl_st20p_tx.c
@@ -106,6 +106,18 @@ static int mtl_st20p_write_header(AVFormatContext* ctx) {
       ops_tx.input_fmt = ST_FRAME_FMT_RGB8;
       ops_tx.transport_fmt = ST20_FMT_RGB_8BIT;
       break;
+    case AV_PIX_FMT_YUV444P10LE:
+      ops_tx.input_fmt = ST_FRAME_FMT_YUV444PLANAR10LE;
+      ops_tx.transport_fmt = ST20_FMT_YUV_444_10BIT;
+      break;
+    case AV_PIX_FMT_GBRP10LE:
+      ops_tx.input_fmt = ST_FRAME_FMT_GBRPLANAR10LE;
+      ops_tx.transport_fmt = ST20_FMT_RGB_10BIT;
+      break;
+    case AV_PIX_FMT_YUV420P:
+      ops_tx.input_fmt = ST_FRAME_FMT_YUV420CUSTOM8;
+      ops_tx.transport_fmt = ST20_FMT_YUV_420_8BIT;
+      break;
     default:
       err(ctx, "%s, unsupported pixel format: %d\n", __func__, s->pixel_format);
       return AVERROR(EINVAL);
-- 
2.34.1

